environment.py

```
@@ -0,0 +1,65 @@
+from typing import Optional
+import numpy as np
+import gymnasium as gym
+from gymnasium.spaces import OneOf, Box, Dict
+import client
+import cv2
+
+
+class TankwarsEnv(gym.Env):
+
+    def __init__(self, client):
+        self.client = client
+        self.observation_space = Box(0, 255, shape=(200, 200, 3), dtype=np.uint8)
+        self.action_space = Dict({
+            "right_engine": Box(-1, 1, shape=(1,), dtype=np.float32),
+            "left_engine": Box(-1, 1, shape=(1,), dtype=np.float32),
+            "fire": OneOf(True, False),
+        })
+
+        def get_state(self, player_id):
+            self.client.request_observation(player_id, self.client.ObservationKind.IMAGE)
+            return self.client.player_state(player_id)
+
+        def get_info(self):
+            pass
+
+        def reset(self, seed=None):
+            # super().reset(seed=seed)
+
+            print("Requesting a player to be controlled")
+            player_id = client.get_player()
+
+            if player_id is None:
+                return print("Failed to receive a player to control, quitting!")
```

This should raise an exception signaling the env could not get a player